e2e/app/element/base-element.ts

```diff
@@ -108,33 +111,32 @@ export default class BaseElement {
 
   /**
    * <pre>
-   *  Check if the element is visible
+   *  Check both boxModel and style for visibility to determine whether element is visible.
    * </pre>
-   * @param {Page} page
-   * @param {ElementHandle} element
    */
   async isVisible(): Promise<boolean> {
+    const computedStyle = async (element) => {
+      return this.page
+        .evaluateHandle((elem) => {
```

Comment:
nit: since you use `async` here, `await` would probably be more natural than promise chaining